### DISCUSSION TITLE
AnnoucementBanner with animation

### DISCUSSION THREAD

**Discussion by kaminski-dev** [tailwindlabs/tailwindcss#11912] [Help] [answered]
Hi, I would like to do an announcement banner where the text is like animated infinitely. 

https://play.tailwindcss.com/SkI4o22quL

Basically I don't have much right now, but it should be like that, that this Hello World will go from right to left infinitely. Only if I hover, it will stop. And for mobile, is there an alternative to hover?

Many thanks in advance for helping me on that!

**Comment by wongjn** [comment, marked as answer]
https://play.tailwindcss.com/QAk8v9Mq0Z

- Custom `animation` values go in `theme.animation` or `theme.extend.animation`, not in `animation`.
- `transform translate-x-[calc(100%+2rem)]` and `translate-x-0` are not valid CSS values for `transform`.
- Have a "clone" of the text element, so that it is seamless when animating.
- Use `animation-play-state` with `hover:` to have the animation pause on hover.

**Comment by kaminski-dev** [comment]
Perfect this helped me a lot.

I have just one question left anf that is that I have a gradient defiend in my tailwindcss config 

      backgroundImage: {
        "golden-gradient":
          "linear-gradient(to right, #C29439, #D9B879, #765713, #FFE69F, #906420, #D9BB7B, #906210)",
      },
I added it to the text and with the option w-full the color of the texted changed, without it it is as it should be. 

Do you know what could be the reason for this? 

https://play.tailwindcss.com/IqaQhNdHnh

Thanks in advance for the answer.

**Reply by wongjn** [reply]
You'd want to use the `class` attribute, not `className`: https://play.tailwindcss.com/IqaQhNdHnh

**Reply by kaminski-dev** [reply]
Ah ok this is the reason why it was not redering, but the gradient changes back to default when I delete the second w-full in the second line! Thank you for your support!
